streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableSourceTest.java

```diff
@@ -85,6 +89,40 @@ public void testKTable() {
             supplier.theCapturedProcessor().processed);
     }
 
+    @Test
+    public void testKTableSourceEmitOnChange() {
```

Comment:
Q: Why do you need the `TopologyTestDriver` here? I see that the other test use it. I guess you could simply instantiate a `KTableSource`, get the processor from it, and test directly the processor.

Reply:
I think that we can just follow the previous test format, so I think there's not too much need to change it at the moment.